File: StockTracker.Api/Models/Stocks.cs
namespace StockTracker.Api.Models;

public class Stock
{
    public int Id { get; set; }
    public string Ticker { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public decimal PurchasePrice { get; set; }
    public decimal Quantity { get; set; }
    public DateTime PurchasedAt { get; set; }
    public string? Notes { get; set; }
}

File: StockTracker.Client/src/api/stocks.ts
import axios from 'axios';

export interface Stock {
  id: number;
  ticker: string;
  name: string;
  purchasePrice: number;
  quantity: number;
  purchasedAt: string;
  notes?: string;
}

const api = axios.create({
  baseURL: '/api',
});

export const getStocks = () => api.get<Stock[]>('/stocks');
export const createStock = (stock: Omit<Stock, 'id'>) => api.post<Stock>('/stocks', stock);
export const deleteStock = (id: number) => api.delete(`/stocks/${id}`);